scripts/teamd_increase_retry_count.py

```diff
@@ -0,0 +1,320 @@
+        ByteField("partner_type", 2),
+        ByteField("partner_length", 20),
+        ShortField("partner_system_priority", 0),
+        MACField("partner_system", None),
+        ShortField("partner_key", 0),
+        ShortField("partner_port_priority", 0),
+        ShortField("partner_port_number", 0),
+        ByteField("partner_state", 0),
+        XStrFixedLenField("partner_reserved", "", 3),
+        ByteField("collector_type", 3),
+        ByteField("collector_length", 16),
+        ShortField("collector_max_delay", 0),
+        XStrFixedLenField("collector_reserved", "", 12),
+        ConditionalField(ByteField("actor_retry_count_type", 0x80), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("actor_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_type", 0x81), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("partner_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ByteField("terminator_type", 0),
+        ByteField("terminator_length", 0),
+        ConditionalField(XStrFixedLenField("reserved", "", 42), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("reserved", "", 50), lambda pkt:pkt.version != 0xf1),
+    ]
+
+split_layers(scapy.contrib.lacp.SlowProtocol, scapy.contrib.lacp.LACP, subtype=1)
+bind_layers(scapy.contrib.lacp.SlowProtocol, LACPRetryCount, subtype=1)
+
+class LacpPacketListenThread(Thread):
+    def __init__(self, port, targetMacAddress, sendReadyEvent):
+        Thread.__init__(self)
+        self.port = port
+        self.targetMacAddress = targetMacAddress
+        self.sendReadyEvent = sendReadyEvent
+        self.detectedNewVersion = False
+
+    def lacpPacketCallback(self, pkt):
+        if pkt["LACPRetryCount"].version == 0xf1:
+            self.detectedNewVersion = True
+        return self.detectedNewVersion
+
+    def run(self):
+        sniff(stop_filter=self.lacpPacketCallback, iface=self.port, filter="ether proto {} and ether src {}".format(LACP_ETHERTYPE, self.targetMacAddress),
+                store=0, timeout=30, started_callback=self.sendReadyEvent.set)
+
+def getPortChannels():
+    applDb = DBConnector("APPL_DB", 0)
+    configDb = DBConnector("CONFIG_DB", 0)
+    portChannelTable = Table(applDb, "LAG_TABLE")
+    portChannels = portChannelTable.getKeys()
+    activePortChannels = []
+    for portChannel in portChannels:
+        state = portChannelTable.get(portChannel)
+        if not state[0]:
```

Comment:
Risk of index error here?

Reply:
I think libswsscommon returns whether the call is a success or failure in the first index of the tuple that's returned, but regardless, I added a check to see if an empty tuple/list was returned.